src/components/modules/caret.ts

```diff
@@ -205,13 +209,13 @@ export default class Caret extends Module {
      * Why not regular .trim():
      *  in case of ' hello |' trim() will also remove space at the beginning, so length will be lower than anchorOffset
      */
-    const rightTrimmedText = lastNode.textContent.replace(/\s+$/, '');
+    const text = ignoreSpaces ? lastNode.textContent.replace(/\s+$/, '') : lastNode.textContent;
```

Comment:
I have discovered a bug on 2.22.2 and I'm looking for hints how to solve it properly and understand the code a bit better. During my research I found this PR and I just want to mention:

In certain scenarios there is a chance that `lastNode` might be `undefined`. In L190 you're checking it correctly but here `if (lastnode && ...` you're not. 

In my scenario I have custom block which which uses onPaste configurations. The Block itself uses `ReactDOM.render()` to offload it into a react component - just in case it's important. 

If my caret is in the middle of the text and I paste my "pattern" then after moving left,up,down,right then I receive several errors related to `isAtEnd()` and `$.isEmpty()` which uses a `node.normalize()` while `node` is `undefined`. Dunno yet why. Trying to identify it on my side.

Reply:
Just an informational comment: After debugging further I realised: After I paste my "onPaste" inside a paragraph and the block is successfully added, then my caret is still at the paragraph-block. But when I start moving up,down,left,right then the result of `this.Editor.BlockManager.currentBlock` returns the new inserted block instead of the position of the caret.

Need to debug further.

Reply:
Why not just stop doing trim or similar to discover if the caret is at the end or start?